---
layout: post
title: Code
cover: code.jpg
categories: posts
---

Throughout the course, we will break out of the lecture to explore concepts more thoroughly through computational analysis. We'll write a wide variety of code covering concepts such as mathematical techniques, stochastic simulations, and image processing. Furthermore, we will work through an experimental data set to quantify gene expression in bacterial cells.

## Introductory Materials
Please work through the following tutorials before the beginning of class. The TAs will hold a special session covering these topics on **Sunday, January 20th** from 16:00 to 17:00 in College A.227. **Note:** *In some browsers (Firefox primarily), the end of each sentence is clipped off, making it difficult to read. If this is occurring, please try using another browser (Chrome, Safari, etc).*

* [Tutorial 0a: Configuring your computer]({{ site.baseurl}}/code/t0a_setting_up_python.html) \| This tutorial will walk you through how to install a Python 3.6 scientific computing environment.

* [Tutorial 0b: Using the Jupyter notebook]({{ site.baseurl}}/code/t0b_jupyter_notebooks.html) \| This tutorial will teach you how to write code and text in Jupyter notebooks for homework submissions.

* [Tutorial 0c: A Primer on Python Syntax]({{ site.baseurl}}/code/t0c_python_syntax_and_plotting.html) \| This tutorial will walk you through the basics of programming in Python.


## Data sets

Please download the following data sets, unzip them, and place them in your `bootcamp/data` folder as described in the [setting up Python]({{ site.baseurl}}/code/t0a_setting_up_python.html) tutorial.

* [Data Set 1](http://www.rpdata.caltech.edu/courses/course_data/ecoli_growth.zip) \| A series of phase contrast and fluorescence images of a growing *E. coli* colony

* [Data Set 2](http://www.rpdata.caltech.edu/courses/course_data/lacI_titration.zip) \| A large image set of *E. coli* strains with varying copy numbers of the lacI repressor molecule.

* [Data Set 3](http://www.rpdata.caltech.edu/courses/course_data/fly_elongation.zip)
    \| An image set of 5' and 3' labeled mRNA expressing in the developing
    *Drosophila* embryo.


## Course Python Utilities
As sometimes syntax can be difficult, we have written a file with a few functions written in Python that will make some of the in-class exercises less cumbersome. Please download them below and place them in your root `bootcamp` folder as described in the [setting up Python]({{ site.baseurl}}/code/t0a_setting_up_python.html) tutorial:

[pboc_utils.py](../../../../code/pboc_utils.py) \| Course utilities.


## Course Exercises
As we go through the course, the code we write in class will be posted here. When possible, extra tutorials with more detail and explanation will be posted as well.

* [Exercise 1]({{ site.baseurl}}/code/forward_euler.html) \| Numerically integrating the differential equation for exponential growth. \[[In class]({{ site.baseurl}}/code/forward_euler_in_class.html)\]

* [Exercise 2]({{ site.baseurl}}/code/bacterial_growth.html) \| Determining bacterial growth rate from microscopy data. \[[Data set](http://rpdata.caltech.edu/courses/course_data/ecoli_growth.zip)\] \[[In class]({{ site.baseurl}}/code/bacterial_growth_in_class.html)\]

* [Exercise 3]({{ site.baseurl}}/code/diffusion_via_coin_flips.html) \| Simulating diffusing particles with coin flips. \[[In class]({{ site.baseurl}}/code/diffusion_via_coin_flips_in_class.html)\]


## Extra exercises
Below are some coding exercises that were not performed in class, but are
relevant to our discussions.
