---
layout: post
title: Readings
cover: readings.jpg
categories: posts
---

## In-Class Readings

Readings mentioned specifically in class will be listed here as the course progresses.

### Monday, January 21

<a href="https://bionumbers.hms.harvard.edu/search.aspx"> Bionumbers website</a>, maintained by Ron Milo's lab

<a href="http://rpdata.caltech.edu/publications/Bar-On_2018.pdf"> The Biomass Distribution on Earth </a> by Bar-On et al. <i>PNAS</i> 2018

<a href="http://www.rpdata.caltech.edu/courses/course_papers/Naturemicrobiology_Wigington_2016.pdf"> Re-examination of the relationship between marine virus and microbial cell abundances </a> by Wigington et al. <i>Nature Microbiology</i> 2016

<a href="https://bmcbiol.biomedcentral.com/articles/10.1186/1741-7007-12-29"> Models in biology: ‘accurate descriptions of our pathetic thinking’ </a> by Jeremy Gunawardena <i>BMC Biology</i> 2014

<a href="http://www.rpdata.caltech.edu/courses/course_papers/NatureBiotech_Schmidt_2016.pdf"> The quantitative and condition-dependent Escherichia coli proteome </a> by Schmidt et al. <i>Nature Biotechnology</i> 2016

### Tuesday, January 22

<br/>

### Wednesday, January 23

<br/>

### Thursday, January 24

<br/>

### Friday, January 25

<br/>

## Supplementary reading

The papers provided here are meant to provide an entry point
into the literature for going more deeply into various topics
covered -in class.   These papers have been picked either
because they provide interesting and provocative experimental
measurements of particular biological phenomena or because they
show how to go about constructing theoretical models in the
physical biology spirit described in the course. The papers
that of most direct relevance to what we will cover in class
are linked on the "Syllabus" part of the website.

### Biology by the numbers
 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Milo%20Cell%20SnapShot%202010.pdf">Uri Moran et al., (2010), <i> SnapShot: Key Numbers in Biology</i>, Cell 141, pp.1262</a> <br/> List of key numbers in biology, such as the quantity and size of cellular components and the rates of cellular processes.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/eddy%20bayesian%20methods.pdf">
Sean Eddy (2004), <i> What is Bayesian statistics?</i> Nature Biotechnology 22, pp.  1177-178. </a><br />
This paper gives a compact but beautiful example of the
power of Bayesian methods for figuring out the
probability ofsome hypothesis given the data.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/First_exposure_Garcia_2007.pdf">
    Hernan Garcia et al. (2007), <i> A First Exposure to Statistical Mechanics for Life Scientists</i>.</a><br />
This paper is a brief introduction to ideas from statistical mechanics that can
be used to analyze a variety of problems in biology.

### The physics of genome management
<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/kornbergStryernar_nucleosome_positioning.pdf">
Roger Kornberg and Lubert Stryer (1988), <i>Statistical distribrutions of nucleosomes: nonrandom locations by a stochastic mechanism,</i> Nucleic Acids Research 16, pp. 6677-90.</a><br />
This paper shows how a simple model of excluded volume predicts how nucleosomes will be organized around promoters.


 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Widom_Segal_2006_Nature.pdf">
Eran Segal et al. (2006), <i> A genomic code for nucleosome positioning,</i> Nature 442, pp. 772-8.</a><br />
This paper describe work aimed at determining genome wide nucleosome positioning preferences.

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Widom_Segal_2009_Nature.pdf">
Noam Kaplan et al. (2009), <i> The DNA-encoded nucleosome organization of a eukaryotic genome,</i> Nature 458, 362-6.</a><br />
This paper describe work aimed at determining genome wide nucleosome positioning preferences.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/PNAS-2010-Wiggins.pdf">
Paul Wiggins et al. (2010),<i> Strong intranucleoid interactions organize the </i>E. coli<i> chromosome into a nucleoid filament,</i> PNAS 107, pp. 4991-5.</a><br />
This paper looks at the spatial organization of the genome in a cell.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Bustamante%20and%20Smith%20Nature%202001.pdf">
Douglas Smith et al. (2001), <i>The bacteriophage Φ29 portal motor can package DNA gains a large internal force,</i> Nature 413, 748-52.</a><br />
In this paper, optical tweezers are used to study the forces needed to package double-stranded DNA into a viral capsid.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Tsankov_PLoS_Biology_2010.pdf">
Alexander Tsankov et al. (2010), <i> THe role of nucleosome positioning in the evolution of gene regulation,</i> PLoS Biology 8, e1000414.</a><br />
This interesting paper examines the genome-wide nucleosome positions in
12 different yeast species. This data provides an excellent jumping off
point for models of nucleosome positioning.

 <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/PolachWidomJMB1995.pdf">
K.J. Plach and J. Widom (1995), <i>Mechanism of protein access to specific DNA sequences in chromatin: A dynic equilibrium model for gene regulation,</i> JMB 254, 130-49.</a><br />
This paper examines how different sites within nucleosomes
grant access to DNA binding proteins and quantifies how this
accessibility depends upon the depth of the sites of interest
within the nucleosome.

### Regulatory biology

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Garcia%20PNAS%202011.pdf">
    Hernan Garcia and Rob Phillips (2011), <i>Quantitative dissection of the simple repression input-output function</i>, PNAS 108(29) pp. 12173-8.</a><br />
This paper demonstrates that thermodynamic models can make
quantitative predictions about the level of gene expression as
a function of repressor copy number and operator strength.

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/bintu%20Curr%20Op%20Gen%20Dev%20Models%202005.pdf">
    Lacramioara Bintu et al. (2005), <i> Transcriptional regulation by the numbers: models,</i> Current Opinions in Genetics and Development 15, pp. 116-24.</a><br />
This paper outlines an approach to creating quantitative models
of gene expression using thermodynamics of the binding of
transcription factors and RNA polymerase to DNA.

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Golding_Cell_2005.pdf">
    Ido Golding (2005), <i>Real-time kinetics of gene activity in individual bacteria,</i> Cell 123, pp. 1025-36.</a> <br />
This paper clearly demonstrates that the process by which mRNA
is produced in the E.coli cell is stochastic in nature. A
surprising observation is that the mRNA distribution is not
Poisson, characterized by bursts in mRNA production. To this
day the source of the stochasticity remains a mystery.

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/BergSourjikPNAS.pdf">
Victor Sourjik and Howard C. Berg (2002), <i>Receptor sensitivity in bacterial chemotaxis</i>, PNAS 99, 132-7</a> and
<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/SourjikBergPNAS-2002-Sourjik-12669-74.pdf">
Victor Sourjik and Howard C. Berg (2002), <i> Binding of the </i> Escherichia coli <i> response regulator CheY to its
target measured in vivo by fluorescence resonance energy
transfer,</i> PNAS 99, pp. 12669-74.</a> <br />
These two papers use the method of FRET to examine the relation
between chemoattractant concentration and the chemical
reactions within cells that control the frequency of tumbles.

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/keymer06.pdf">
    Juan Keymer et al. (2006), <i> Chemosensing in </i>Escherichia coli<i>: Two regimes of two-state receptors,</i> PNAS 103, pp. 1786-91.</a>
and
  <a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Tu-PNAS-2005-Mello-17354-9.pdf">
Bernardo A. Mello and Yuhai Tu (2005), <i>An allosteric model
  for heterogeneous receptor complexes: Understanding
  bacterial chemotaxis responses to multiple stimuli</i>,
PNAS 102(48), pp. 1753-9.</a><br />
Theory of chemotaxis: These two papers show how simple ideas
from equilibrium statistical mechanics can be used to
understand the chemotactic response of E. coli to different
concentrations of chemoattractant.

### Pattern formation in biology

<a href="http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/gregor+al_05.pdf">
Thomas Gregor et al. (2005), <i>Diffusion and scaling during early embryonic pattern formation,</i> PNAS 102, pp. 18403-7.</a><br />
Quantitative analysis reveals that diffusion based mechanism
cannot account for morphogen gradient scaling in early embryos
across closely related fly species.

### Evolution
[Theodsius Dobzhansky (1973), Nothing in biology makes sense except in the light of evolution, The American Biology Teacher 3, 125-9](http://www.rpdata.caltech.edu/courses/PBoC_CSHL_2015/files_2015/articles/Dobzhansky_1973.pdf)
This classic article makes a compelling case for the primacy of evolution in the study of biology.
